### PR TITLE
Document VHost Regexp #344

### PR DESCRIPTION
**Pull request by rymsha**

### CODE REVIEW

File: docs/deployment/vhosts.adoc [hunk 1]
```diff
@@ -19,13 +19,76 @@ By using vhosts, traffic for specific domains and url patterns may be routed dir
...
+.image:xp-780.svg[XP 7.8.0,opts=inline] host mapping match both `www.example.com` and `example.com` hosts
+[source,properties]
+----
+mapping.example.host = www.example.com example.com
+----
+
+.image:xp-780.svg[XP 7.8.0,opts=inline] host mapping will match any sub-domain: `1.example.com`, `2.example.com` hosts (but not sub-sub-domains)
+[source,properties]
+----
+mapping.example.host = ~[^.]\\.example\\.com
```
Comment (anatol-sialitski) [inline docs/deployment/vhosts.adoc:53]: should be `~[^.]+\\.example\\.com`

File: docs/deployment/vhosts.adoc [hunk 2]
```diff
@@ -19,13 +19,76 @@ By using vhosts, traffic for specific domains and url patterns may be routed dir
...
+.source mapping matches any request-uri that equals to `/hello` or starts with `/hello/`
+[source,properties]
+----
+mapping.example.source = /hello
+----
+
+If multiple VHosts match the request, disambiguation rules apply in the following order:
+
+- image:xp-780.svg[XP 7.8.0,opts=inline] VHost with lower `order` value is preferred.
+If `order` is not specified, order is considered _lest_ preferable (`Integer.MAX_VALUE`).
```
[the remark this answers is not in the document]
Comment (rymsha) [reply docs/deployment/vhosts.adoc:74]: least is correct

File: docs/deployment/vhosts.adoc [hunk 3]
```diff
@@ -19,13 +19,76 @@ By using vhosts, traffic for specific domains and url patterns may be routed dir
...
+- VHost with the longest `source` value is preferred.
```
[the remark this answers is not in the document]
Comment (rymsha) [reply docs/deployment/vhosts.adoc:75]: that is a bug in code. and spec

[the remark this answers is not in the document]
Comment (anatol-sialitski) [reply docs/deployment/vhosts.adoc:75]: it was fixed in https://github.com/enonic/xp/pull/9141

**Review by alansemenov — APPROVED** [review]

**Pull request merged**